airflow/providers/databricks/hooks/databricks_sql.py

```diff
@@ -241,13 +267,23 @@ def run(
         else:
             return results
 
-    @staticmethod
-    def _make_serializable(result):
-        """Transform the databricks Row objects into JSON-serializable lists."""
-        if isinstance(result, list):
-            return [list(row) for row in result]
-        elif isinstance(result, Row):
-            return list(result)
+    def _make_common_data_structure(
+        self, result: list[T] | T | None
+    ) -> list[T] | T | list[tuple] | tuple | None:
+        """Transform the databricks Row objects into namedtuple."""
+        # Below ignored lines respect namedtuple docstring, but mypy do not support dynamically
+        # instantiated namedtuple, and will never do: https://github.com/python/mypy/issues/848
+        if self.return_tuple:
+            if isinstance(result, list) and all(isinstance(item, Row) for item in result):
```

Comment:
Potentially you are checking thousands of rows here and `isinstance` is quite slow as it needs to walk the inheritance tree. Is there a way to check a attribute(s) to verify you are dealing with a Row?

Reply:
What would be your suggestion here @bolkedebruin ? I guess it is not blocking but simple optimisation suggestion? If so - is that fine if we separate it as a follow up issue to optimize it? I guess this is not something that BLOCKS this pr? 

I think we can definitely optimise it in the future, however, again - perfect is the enemy of good and it's holding back release of a new waves of providers . My suggestion is to get it out now and separate it as a follow-up PR.

Is that OK @bolkedebruin ?

Reply:
an `isinstance` check can take between 15%-60% more time than an attribute check. This change affects current operations, so maybe this is an pre-mature optimization or maybe we do want to apply it. I leave this to @Joffreybvn as he is using it I assume.